tools/topology/m4/dcblock_coef_default.m4

```diff
@@ -0,0 +1,24 @@
+CONTROLBYTES_PRIV(DCBLOCK_priv,
+`       bytes "0x53,0x4f,0x46,0x00,' dnl magic number 'S' 'O' 'F' '\0'
```

Comment:
I am down for tokens, unless you or @singalsu think there may be additional configuration parameters in the future that may need to be added (personnally I don't think there will be.) Can you point to an example of token usage currently?

Reply:
This filter currently does not fully remove the DC offset at the beginning, which can cause a pop to play through the speaker for the first few frames. 

We would need to apply a mix-in ramp to get rid of those audio discontinuities. This would require an extra parameter to be added to the configuration. 

In this case, would you still suggest to use tokens, or should we keep using a blob?

Reply:
We use for capture pipelines a volume component fade-in ramp that does this. Setting it to slightly long than default 250 ms helps, I think we use 400 ms ramp. Currently volume supports only linear ramp type so the conceal effect is quite short.

Reply:
@cujomalainey @sebcarlucci Have you considered adding parameter for fixed gain for this component? It could be single or channel specific, though I feel channel specific gain would be better to handle in equalizer if there's systematic level difference in transducers.

We have set for some platforms which should come out of the box with suitable capture sensitivity +20 dB gain at DC blocking high-pass filter for DMIC. Since the volume component defaults to 0 dB and apparently also it's not in topology and UCM possible (@juimonen is still a limitation?) to define a default gain the fixed gain at DC blocker high-pass has been a solution, though not elegant.

Reply:
For the ramp time ok I think we will remove it and assume there is a volume component in the pipeline to handle this for us.

I like that idea, it would be good for equalizing the mics, but I think that should be something we should land in a separate PR.

Reply:
@singalsu and @juimonen are the structs for the binary controls published to userspace? I am thinking of how controling the interface will work dynamically from a program with a machine that does not have octave/matlab on it.

Reply:
Yes, there's SOF/src/include/user, and e.g. file eq.h. In DC blocker case with the first order filter it's possible to compute the filter coefficient with closed form equation from e.g. requested cutoff frequency.

Higher order IIR filters would require polynomial solve etc. libraries and FIR would need FFT/IFFT and window function. Such libraries do exist as open source, likely GPL (e.g. FFTW), so it works as long as the tool is compatible with license.